### PR TITLE
original v9 changes 

### PR DESCRIPTION
**Pull request by sandgraham**
#### Description

This PR updates styles across Base Web components.

- Update base color primitives
  - Unify light and dark primitives (no difference anymore)
- Update buttons to gray scale
  - Deprecate `minimal` button for removal in future major
- Update button group to use new button styles
- Update controls to use black highlights
  - input
  - select
  - checkbox
  - radio
  - slider
  - tabs
  - drag and drop
  - date-picker
  - progress steps
- Update typographic scale
- Added a codemod to `baseui-codemods`
  - Maps usage of mono colors
  - Maps usage of font values
  - Maps typography components

### CODE REVIEW

**Comment by gergelyke** [comment]
should we combine this with the black controls, so folks get them in one go? that may be less confusing?

**Comment by sandgraham** [comment]
Ah - same PR or is releasing a minor with both ok?

**Comment by gergelyke** [comment]
probably I'd do the same pr, so we can release the master - also, let's announce this tomorrow in the newsletter so folks will have some headsup

**Comment by sandgraham** [comment]
Ok- might take a couple days to get the new focus state stuff in.

File: documentation-site/pages/blog/style-updates/metadata.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "author": "Graham Murdoch",
+  "authorLink": "https://github.com/sandgraham",
+  "title": "Color and Typography Updates in V8.17",
```
Comment (gergelyke) [inline documentation-site/pages/blog/style-updates/metadata.json]: nit: to be consistent with the rest of the changeset, can we use `v8.17`?

File: documentation-site/pages/blog/style-updates/index.mdx [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+The base color primitives for both the default light and dark theme have been adjusted. These have been changed to improve the contrast ratios across our color palette.
+
+The changes in most of the color ways are quite subtle, but the gray (mono) palette has shifted more noticeably. Here are the updated gray scale values:
+
+### Light Theme Changes
+
+<Table
+  columns={['Primitive', 'Old', 'New']}
+  data={[
+    ['mono50', '-', '#F6F6F6'],
```
Comment (gergelyke) [inline documentation-site/pages/blog/style-updates/index.mdx]: may be an overkill, but I wonder, if we could add a small rectangle next o the color codes, so it's easier to understand?

**Comment by gergelyke** [comment]
looks great! one minor comment - should we make the component type (Inputs) smaller, so it becomes smaller than the main menu category(Components)?

Screen Shot 2019-08-15 at 11 28 50 AM

File: documentation-site/pages/blog/style-updates/index.mdx [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+<Meta data={metadata} />
+
+<BlogImage
+  src="/static/images/blog/style-updates/baseblack.png"
+  alt="updated base web components with black highlights"
+  caption="Base Web has an updated look"
+/>
+
+## Migration Guide
```
Comment (gergelyke) [inline documentation-site/pages/blog/style-updates/index.mdx]: should we call this upgrade guide? ideally, this change shouldn't require code changes, right?

File: documentation-site/pages/blog/style-updates/index.mdx [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+
+After bumping to `v8.17`, your app will look a little different. Grays will be darker and type will be larger. Input highlights and buttons will be black rather than blue. _Don't panic!_ If you address each of the points below your app will look right in no time!
+
+- Replace direct usage of gray scale (`$theme.colors.mono100` for example) with new values. Make sure to look for references in `Block` as well. [Read more 📖](#light-theme-mapping)
+- Replace direct usage of typographic scale (`$theme.typography.font100` for example) with new values. Make sure to look for references in `Block` as well. [Read more 📖](#typography-mapping)
```
Comment (gergelyke) [inline documentation-site/pages/blog/style-updates/index.mdx]: do we want people to replace these values? if we think it's a must, should we release as a major?

Comment (abmai) [reply documentation-site/pages/blog/style-updates/index.mdx]: +1 I think this should be a major version bump. It's a breaking change unless manual migration (i.e. follow the upgrade guide) is done.

**Comment by sandgraham** [comment]
This will be picked up in a new PR.

**Pull request closed without merging**